### DISCUSSION TITLE
save jobs separately and combine it

### DISCUSSION THREAD

**Discussion by Niccolum** [community/community#25093] [Actions]
<p>Is there a way to store the jobs separately, to combine them and (most importantly) do something like<br>
<a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idneeds" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idneeds</a><br>
For instance:<br>
In the example <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/configuring-a-workflow#referencing-actions-in-your-workflow" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/configuring-a-workflow#referencing-actions-in-your-workflow</a><br>
It is written about what interests me, but there is no fine-tuning. Tell me, is there such an opportunity? Scrolling through all the documentation, I not found a suitable link.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Jobs are defined in workflow, however there’s no dependency between workflows. Hence, if store jobs separately, it’s NOT supported to combine them and make dependency(like ‘needs’) between them.</p>
<p>As an alternative, with personal access token as encrypted secrets, you can trigger a workflow(and jobs in it) by another one. For example, commit some code in 1st workflow with personal access token as encrypted secrets, set the event to ‘push’ to trigger the 2nd workflow.</p>
<p>Or you can convert the jobs to actions and involve them in one workflow accordingly.</p>

**Comment by Niccolum** [comment]
<p>Thanks for your answer!<br>
Then I had new questions:</p>
<ol>
<li>If i trigger new (second) workflow - old workflow (first) will not wait for the completion of the second and will succeed despite the result of the second? If I understand correctly - this is not quite what I need.</li>
<li>How to convert jobs to actions? Perhaps due to carelessness, I could not find a suitable section with examples in the documentation.<br>
Thanks</li>
</ol>

**Comment by weide-zhou** [comment]
<p>Thanks for your mark!</p>
<p>For your question :</p>
<ol>
<li>Yes, you understand correctly. The two workflows are separated, the 1st workflow creates (eg) code submit and satisfies the trigger event of 2nd workflow.</li>
<li>You can create own actions which combines of the execution steps in the job. Please check the action <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/building-actions" rel="nofollow noopener">doc</a> for more details.</li>
</ol>
<p>In your case, if you would like 1st workflow wait for completion of 2nd workflow and check the result of it accordingly, probably put the jobs in one workflow, execute them in sequence is a not a bad choice.</p>
